### ISSUE TITLE
Global middleware fails to call next() on a route with route-specific middleware(s) if 2nd param is a variable to an options object

### ISSUE THREAD

**Issue by BobbyWibowo** [kartikk221/hyper-express#102]
HyperExpress v6.4.4 on Node 16.15.1

```
TypeError: middleware.middleware is not a function
    at Route.handle (/home/bobby/src/safe.fiery.me/node_modules/hyper-express/src/components/router/Route.js:89:37)
    at iterator (/home/bobby/src/safe.fiery.me/node_modules/hyper-express/src/components/router/Route.js:79:22)
    at /home/bobby/src/safe.fiery.me/controllers/middlewares/RateLimiter.js:47:16
```
That error will throw on every requests, assuming the following scenario:

`RateLimiter.js` is a synchronous global middleware that will call `next()` if it passes,
and the route is specified like so:
```js
const routes = new Router()
const options = { max_body_length: 100 * 1e6 }
routes.post('/upload', options, auth.optionalUser, upload.upload)
routes.post('/upload/:albumid', options, auth.optionalUser, upload.upload)
```
which, according to https://github.com/kartikk221/hyper-express/blob/6.4.4/docs/Router.md#router-instance-methods, should conform to the following overload type:
```
any(String: pattern, Object: options, Function | Function[]: ...middleware, Function: handler)
```
and `auth.optionalUser` is a generic middleware (sync or async does not appear to matter)

I notice that it does not throw if the object is specified directly into the 2nd param, instead of a variable pointing to it
```js
routes.post('/upload', { max_body_length: 100 * 1e6 }, auth.optionalUser, upload.upload)
routes.post('/upload/:albumid', { max_body_length: 100 * 1e6 }, auth.optionalUser, upload.upload)
```

The same error would be thrown if route-specific middleware is specified in the object var
```js
const options = {
  max_body_length: 100 * 1e6,
  middlewares: [auth.optionalUser]
}
routes.post('/upload', options, upload.upload)
routes.post('/upload/:albumid', options, upload.upload)
```

I legitimately don't understand why

---

**EDIT:** It appears that the fact that there is another route referencing to the options variable is relevant
If I don't have the 2nd route `/upload/:albumid`, it just works

I'm guessing the provided object gets tampered with when used on a route, and it ends up in a way that is then no good for use by other routes
Doing the following makes it behave
```js
routes.post('/upload', Object.assign({}, options), upload.upload)
routes.post('/upload/:albumid', Object.assign({}, options), upload.upload)
```

If it's for performance reason (?), I think it's sufficient to just note the behavior in the docs

**Comment by BobbyWibowo** [comment]
I cannot replicate this behavior on HyperExpress v6.4.2

**Comment by kartikk221** [comment]
This bug has been fixed in v6.4.5, see If that resolves your issue. Your assumption was correct, this bug would occur because HyperExpress calculates and caches all middlewares each route has to go through into the `options.middleware` property internally for faster runtime performance as we only have to chain through a 1D array before calling the route handler. The problem was that the object would be directly modified, so I have now added a line to make an `Object.assign` copy of the provided options object and also added a test to ensure the provided options object remains immutable after route creation.
